scripts/tutorial-uploader/lib/api.test.ts

```diff
@@ -0,0 +1,151 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { describe, expect, test } from "@jest/globals";
+import { readItem } from "@directus/sdk";
+
+import { API } from "./api";
+import { type LocalTutorialData } from "./local-tutorial-data";
+
+/* Create test data */
+const MOCK_TUTORIAL: LocalTutorialData = {
+  title: "Mock tutorial for testing",
+  short_description: "A mock tutorial for testing with Jest",
+  slug: "mock-tutorial",
+  status: "published",
+  local_path: "scripts/tutorial-uploader/lib/test-data/simple-tutorial",
+  category: "Workflow example",
+  topics: [],
+  reading_time: 50,
+  catalog_featured: false,
+};
+
+/* Just to be sure */
+if (/learning-api\.quantum\.ibm\.com/.test(process.env.LEARNING_API_URL!)) {
+  throw new Error(
+    "Tried to run tests against production! Set the env var LEARNING_API_URL to either staging or local (see tutorial-uploader/README.md)",
+  );
+}
+
+describe("Tutorial uploader API", () => {
+  const api = new API();
+
+  test("upload new tutorial", async () => {
+    expect(await api.getTutorialIdBySlug(MOCK_TUTORIAL.slug)).toBeNull();
+
+    await api.upsertTutorial(MOCK_TUTORIAL);
+    const tutorialId = await api.getTutorialIdBySlug(MOCK_TUTORIAL.slug);
+    expect(tutorialId).toBeTruthy();
+
+    const retrievedTutorial = await api.client.request(
+      // @ts-ignore
+      readItem("tutorials", tutorialId, { fields: ["*", "translations.*"] }),
```

Comment:
With this `readItem`, are you trying to read nested fields? If that's the case, FYI (probably you already know), I found this alternative [syntaxis](https://docs.directus.io/guides/sdk/getting-started.html#read-nested-fields). Maybe this could help with the `@ts-ignore` :thinking:. If that doesn't help or the `readItem` had another purpose, I still like how you have made it work. I didn't know about `@ts-ignore` before :rocket:

Reply:
Awesome! Yes always best to fix the type rather than ignore when possible. That fixed all but one.